### PR TITLE
FOGL-5872: backup upload support

### PR DESCRIPTION
**Pull request by mshariq-nerd**
Signed-off-by: Mohd. Shariq <mohd.shariq@nerdapplabs.com>

JIRA logged for api improvements 
https://scaledb.atlassian.net/browse/FOGL-5913

### CODE REVIEW

File: src/app/components/common/file-upload-modal/file-upload-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<div id="fileUpload" class="modal">
+  <div class="modal-background"></div>
+  <div class="modal-card">
+    <header class="modal-card-head">
+      <p class="modal-card-title is-size-6">Upload Backup</p>
```
Comment (praveen-garg) [inline src/app/components/common/file-upload-modal/file-upload-modal.component.html:5]: if this is a common/generic modal then,

title should be passed as in input?

File: src/app/components/common/file-upload-modal/file-upload-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    </header>
+    <section class="modal-card-body">
+      <div class="file has-name is-boxed is-small is-centered">
+        <label class="file-label">
+          <input class="file-input" type="file" name="resume" (change)="attachBackupFile($event)">
```
Comment (praveen-garg) [inline src/app/components/common/file-upload-modal/file-upload-modal.component.html:10]: name resume? change binding should not be hardcoded for generic component

File: src/app/components/common/file-upload-modal/file-upload-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          <span class="file-name">
+            {{fileName}}
+          </span>
+        </label>
+      </div>
+    </section>
+
+    <footer class="modal-card-foot">
+      <button class="button is-small" (click)="formReset()">Cancel</button>
+      <button [disabled]="!fileName || progress || !file"  class="button is-small is-info" (click)="uploadBackup()">Upload</button>
```
Comment (praveen-garg) [inline src/app/components/common/file-upload-modal/file-upload-modal.component.html:28]: click action should not be hard coded/specific to component but caller for generic upload

File: src/app/components/common/file-upload-modal/file-upload-modal.component.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      <progress *ngIf="progress" class="progress is-small is-primary" max="100">15%</progress>
```
Comment (praveen-garg) [inline src/app/components/common/file-upload-modal/file-upload-modal.component.html:29]: what is 15% here?

**Review by praveen-garg — DISMISSED** [review]
Esc works? 
and Close and reopen reset the form?
and should we have a x button (same action as on Cancel) on header top right?

File: src/app/components/common/file-upload-modal/file-upload-modal.component.html (hunk 4 above)
Comment (mshariq-nerd) [reply src/app/components/common/file-upload-modal/file-upload-modal.component.html]: Removed

**Review by praveen-garg — APPROVED** [review]

**Review by monika-Dianomic — APPROVED** [review]

File: src/app/components/common/file-upload-modal/file-upload-modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          this.ngProgress.done();
+          this.progress = false;
+          this.formReset();
+          if (error.status === 0) {
+            console.log('service down ', error);
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        }
+      );
```
Comment (praveen-garg) [inline src/app/components/common/file-upload-modal/file-upload-modal.component.ts:68]: please keep the statements order consistent (same as in success block)

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**